powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/persistence/BasePersistenceStore.java

```diff
@@ -0,0 +1,350 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.idempotency.persistence;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.burt.jmespath.Expression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.utils.StringUtils;
+import software.amazon.lambda.powertools.idempotency.Constants;
+import software.amazon.lambda.powertools.idempotency.IdempotencyConfig;
+import software.amazon.lambda.powertools.idempotency.exceptions.IdempotencyItemAlreadyExistsException;
+import software.amazon.lambda.powertools.idempotency.exceptions.IdempotencyItemNotFoundException;
+import software.amazon.lambda.powertools.idempotency.exceptions.IdempotencyKeyException;
+import software.amazon.lambda.powertools.idempotency.exceptions.IdempotencyValidationException;
+import software.amazon.lambda.powertools.idempotency.persistence.cache.LRUCache;
+import software.amazon.lambda.powertools.utilities.JsonConfig;
+
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Persistence layer that will store the idempotency result.
+ * Base implementation. See {@link DynamoDBPersistenceStore} for an implementation (default one)
+ * Extends this class to use your own implementation (DocumentDB, Elasticache, ...)
+ */
+public abstract class BasePersistenceStore implements PersistenceStore {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BasePersistenceStore.class);
+
+    private String functionName = "";
+    private boolean configured = false;
+    private int expirationInSeconds = 60 * 60; // 1 hour default
+    private boolean useLocalCache = false;
+    private LRUCache<String, DataRecord> cache;
+    private String eventKeyJMESPath;
+    private Expression<JsonNode> eventKeyCompiledJMESPath;
+    protected boolean payloadValidationEnabled = false;
+    private Expression<JsonNode> validationKeyJMESPath;
+    private boolean throwOnNoIdempotencyKey = false;
+    private MessageDigest hashAlgorithm;
+
+    /**
+     * Initialize the base persistence layer from the configuration settings
+     *
+     * @param config       Idempotency configuration settings
+     * @param functionName The name of the function being decorated
+     */
+    public void configure(IdempotencyConfig config, String functionName) {
+        String funcEnv = System.getenv(Constants.LAMBDA_FUNCTION_NAME_ENV);
+        this.functionName = funcEnv != null ? funcEnv : "test-func";
```

Comment:
This is the same as python. And it will probably be only for tests (local) as the LAMBDA_FUNCTION_NAME is defined on Lambda. Fill free to propose something else... "testFunction" sounds more java if you want.

Reply:
Replacing with testFunction